Shared/Extensions/JellyfinAPI/BaseItemDto+VideoPlayerViewModel.swift

```diff
@@ -41,8 +41,58 @@ extension BaseItemDto {
 
         guard let matchingMediaSource = response.value.mediaSources?
             .first(where: { $0.eTag == mediaSource.eTag && $0.id == mediaSource.id })
-        else { throw JellyfinAPIError("Matching media source not in playback info") }
+        else {
+            throw JellyfinAPIError("Matching media source not in playback info")
+        }
 
         return try matchingMediaSource.videoPlayerViewModel(with: self, playSessionID: response.value.playSessionID!)
     }
+
+    func liveVideoPlayerViewModel(with mediaSource: MediaSourceInfo) async throws -> VideoPlayerViewModel {
+
+        let builder = DeviceProfileBuilder()
+        // TODO: fix bitrate settings
+        let tempOverkillBitrate = 360_000_000
+        builder.setMaxBitrate(bitrate: tempOverkillBitrate)
+        let profile = builder.buildProfile()
+
+        let userSession = Container.userSession.callAsFunction()
+
+        let playbackInfo = PlaybackInfoDto(deviceProfile: profile)
+        let playbackInfoParameters = Paths.GetPostedPlaybackInfoParameters(
+            userID: userSession.user.id,
+            maxStreamingBitrate: tempOverkillBitrate
+        )
+
+        let request = Paths.getPostedPlaybackInfo(
+            itemID: self.id!,
+            parameters: playbackInfoParameters,
+            playbackInfo
+        )
+
+        let response = try await userSession.client.send(request)
+
+        var matchingMediaSource: MediaSourceInfo?
+        if let responseMediaSources = response.value.mediaSources {
+            for responseMediaSource in responseMediaSources {
+                if let openToken = responseMediaSource.openToken, let mediaSourceId = mediaSource.id {
+                    if openToken.contains(mediaSourceId) {
+                        matchingMediaSource = responseMediaSource
+                    }
+                }
+            }
+            if matchingMediaSource == nil && !responseMediaSources.isEmpty {
+                // Didn't find a match, but maybe we can just grab the first item in the response
+                matchingMediaSource = responseMediaSources.first
```

Comment:
@sy6sy2 When you get a chance, can you test this latest change? Let's see if using the first item in the response works in your case.

Reply:
It works! Thank you for the fix!
Feel free to ask you want me to try other changes later.